dataquality/integrations/torch_semantic_segmentation.py

```diff
@@ -377,16 +388,26 @@ def get_argmax_probs(
         preds = F.interpolate(preds, size=input_shape, mode="bilinear")
 
         # checks whether the model is (n, classes, w, h), or (n, w, h, classes)
-        if preds.shape[1] == self.number_classes:
+        # takes the max in case of binary classification
+        if max(preds.shape[1], 2) == self.number_classes:
             preds = preds.permute(0, 2, 3, 1)
         assert (
-            preds.shape[-1] == self.number_classes
+            max(preds.shape[3], 2) == self.number_classes
         ), "The model output shape is not as expected. \
                 Expected classes to be in last dimension"
 
-        argmax = (preds.clone().argmax(dim=-1)).cpu()
         logits = preds.clone()  # (bs, w, h, classes)
-        probs = (torch.nn.Softmax(dim=-1)(logits)).cpu()
+        if preds.shape[3] > 1:
+            probs = (torch.nn.Softmax(dim=-1)(logits)).cpu()
+        else:
+            probs = (torch.nn.Sigmoid()(logits)).cpu()
+
+        # expands the binary classification to a 2 channel tensor
+        if probs.shape[3] == 1:
+            probs = self.expand_binary_classification(probs)
```

Comment:
We can move this above, right?
```suggestion
        else:
            probs = (torch.nn.Sigmoid()(logits)).cpu()
            # expands the binary classification to a 2 channel tensor
            probs = self.expand_binary_classification(probs)
```